src/simplify_algebra.cpp

```diff
@@ -1081,6 +1081,46 @@ struct find_splits
         return true;
     }
 
+    int get_binary_op_split_idx(std::vector<instruction_ref> group,
+                                std::vector<instruction_ref> splits) const
+    {
+        auto first_group_inputs = group.front()->inputs();
+        auto arg_it =
+            std::find_if(first_group_inputs.begin(), first_group_inputs.end(), [&](auto i) {
+                return std::find(splits.begin(), splits.end(), i) != splits.end();
+            });
+        auto split_idx = arg_it - first_group_inputs.begin();
+
+        // All splits are at the same input index
+        if(std::all_of(group.begin() + 1, group.end(), [&](auto i) {
+               auto split_idx_input = i->inputs().at(split_idx);
+               return std::find(splits.begin(), splits.end(), split_idx_input) != splits.end();
+           }))
+            return split_idx;
+
+        return -1;
+    }
+
+    void align_commutative_op_args(module& m,
+                                   std::vector<instruction_ref> group,
+                                   std::vector<instruction_ref> splits,
+                                   size_t split_idx) const
+    {
+        auto group_op = group.front()->get_operator();
+        assert(std::all_of(
+            group.begin(), group.end(), [&](auto i) { return i->get_operator() == group_op; }));
+
+        for(auto i : group)
+        {
+            if(std::find(splits.begin(), splits.end(), i->inputs().at(split_idx)) == splits.end())
+            {
+                auto args = i->inputs();
+                std::reverse(args.begin(), args.end());
```

Comment:
reverse trick only works if it binary op

Reply:
you can put an assert that `assert(args.size() == 2)`

Reply:
Ya I can add an assert. The binary op assumption is made because its called from if branch that checks `start->inputs().size() == 2`